pkg/oidctest/issuer.go

```diff
@@ -64,11 +73,85 @@ func NewIssuer(t *testing.T) (jose.Signer, string) {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 		}
 	})
-	oidcServer := httptest.NewServer(oidcMux)
-	t.Cleanup(oidcServer.Close)
 
-	// Setup the testIssuer, so everything uses the right URL.
-	testIssuer = &oidcServer.URL
+	// code stores information sent inside the `code` OAuth parameter.
+	type code struct {
+		ClientID string `json:"client_id"`
+		Nonce    string `json:"nonce"`
+	}
+
+	oidcMux.HandleFunc("/authz", func(w http.ResponseWriter, r *http.Request) {
+		t.Log("Handling request for authz.")
+		redirectURL, err := url.Parse(r.URL.Query().Get("redirect_uri"))
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
I think in all these .Error cases we should return since that doesn't exit the request? Maybe I'm reading the docs wrong howevs?

```
// Error replies to the request with the specified error message and HTTP code.
// It does not otherwise end the request; the caller should ensure no further
// writes are done to w.
// The error message should be plain text.
```